pkg/model/api/v1/role.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	modelAPI "github.com/perses/perses/pkg/model/api"
+)
+
+type RoleInterface interface {
+	GetMetadata() modelAPI.Metadata
+	GetRoleSpec() RoleSpec
+}
+
+type ActionKind string
+
+const (
+	KindWildcard ActionKind = "*"
+	KindRead     ActionKind = "read"
+	KindCreate   ActionKind = "create"
+	KindUpdate   ActionKind = "update"
+	KindDelete   ActionKind = "delete"
+)
+
+type Permission struct {
+	// Action of the permission (read, create, update, delete, ...)
+	Action ActionKind `json:"action" yaml:"action"`
+	// The list of kind targeted by the permission. For example: `Datasource`, `Dashboard`, ...
+	// With Role, you can't target global kinds
+	Scopes []Kind `json:"scopes" yaml:"scopes"`
+}
+
+type RoleSpec struct {
+	// List of permissions owned by the role
+	Permissions []Permission `json:"permissions" yaml:"permissions"`
+}
+
+// GlobalRole is the struct representing the role shared to everybody.
+type GlobalRole struct {
+	Kind     Kind     `json:"kind" yaml:"kind"`
+	Metadata Metadata `json:"metadata" yaml:"metadata"`
+	Spec     RoleSpec `json:"spec" yaml:"spec"`
+}
+
+func (g *GlobalRole) UnmarshalJSON(data []byte) error {
+	var tmp GlobalRole
+	type plain GlobalRole
+	if err := json.Unmarshal(data, (*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*g = tmp
+	return nil
+}
+
+func (g *GlobalRole) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmp GlobalRole
+	type plain GlobalRole
+	if err := unmarshal((*plain)(&tmp)); err != nil {
+		return err
+	}
+	if err := (&tmp).validate(); err != nil {
+		return err
+	}
+	*g = tmp
+	return nil
+}
+
+func (g *GlobalRole) validate() error {
+	if g.Kind != KindGlobalRole {
+		return fmt.Errorf("invalid kind: %q for a GlobalRole type", g.Kind)
+	}
+	if reflect.DeepEqual(g.Spec, RoleSpec{}) {
```

Comment:
```suggestion
if g.Spec == RoleSpec{} {
}
```
that should be simple as that.

Reply:
I copy/paste from datasource/dashboard validation. If I recall correctly it was for checking that spec was correctly set